rllib/utils/actor_manager.py

```diff
@@ -199,84 +248,85 @@ def num_actors(self) -> int:
     @DeveloperAPI
     def num_healthy_actors(self) -> int:
         """Return the number of healthy remote actors."""
-        return sum(self.__remote_actor_states)
+        return sum(
+            [1 for s in self.__remote_actor_states.values() if s.is_healthy]
+        )
 
     @DeveloperAPI
     def num_outstanding_async_reqs(self) -> int:
         """Return the number of outstanding async requests."""
-        return len(self.__in_flight_req_to_actor_idx)
+        return len(self.__in_flight_req_to_actor_id)
 
     @DeveloperAPI
-    def is_actor_healthy(self, idx: int) -> bool:
+    def is_actor_healthy(self, actor_id: int) -> bool:
         """Whether a remote actor is in healthy state.
 
         Args:
-            idx: Index of the remote actor.
+            actor_id: ID of the remote actor.
 
         Returns:
             True if the actor is healthy, False otherwise.
         """
-        return self.__remote_actor_states[idx]
+        return self.__remote_actor_states[actor_id].is_healthy
 
     @DeveloperAPI
-    def set_actor_state(self, idx: int, healthy: bool) -> None:
+    def set_actor_state(self, actor_id: int, healthy: bool) -> None:
         """Update activate state for a specific remote actor.
 
         Args:
-            idx: Index of the remote actor.
+            actor_id: ID of the remote actor.
             healthy: Whether the remote actor is healthy.
         """
-        self.__remote_actor_states[idx] = healthy
+        self.__remote_actor_states[actor_id].is_healthy = healthy
 
     @DeveloperAPI
     def clear(self):
         """Clean up managed actors."""
-        while self.__actors:
-            del self.__actors[0]
+        for actor in self.__actors.values():
```

Comment:
Do you unit-test clear? Can you del a value of a dict without deling the key? why not loop through all the keys and do del self.__actors[key]?

Reply:
yeah it works. I added it to all the unit test cases.
you can't del keys while looping through a dict though.
```
>>> for k in a.keys(): del a[k]
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: dictionary changed size during iteration
>>> 
```
deletion is always tricky. you need to make sure you are not holding an iterator while deleting things.

Reply:
wait, what? this works for me?
```
>>> a = {'a': 1, 'b': 2}
>>> for k in a.keys(): del a[k]
```

Reply:
what?! let me double check.